Source/pack.cpp

```diff
@@ -207,109 +236,168 @@ void UnPackItem(const ItemPack &packedItem, const Player &player, Item &item, bo
-	if (!InDungeonBounds(position)) {
+	Point position { packed.px, packed.py };
+
+	player = {};
+	player._pLevel = clamp<int8_t>(packed.pLevel, 1, MaxCharacterLevel);
+	player._pMaxHPBase = SDL_SwapLE32(packed.pMaxHPBase);
+	player._pHPBase = SDL_SwapLE32(packed.pHPBase);
+	player._pHPBase = clamp<int32_t>(player._pHPBase, 0, player._pMaxHPBase);
+	player._pMaxHP = player._pMaxHPBase;
+	player._pHitPoints = player._pHPBase;
+	player.position.tile = position;
+	player.position.future = position;
+	player.setLevel(clamp<int8_t>(packed.plrlevel, 0, NUMLEVELS));
+
+	player._pClass = static_cast<HeroClass>(clamp<uint8_t>(packed.pClass, 0, enum_size<HeroClass>::value - 1));
+
+	ClrPlrPath(player);
+	player.destAction = ACTION_NONE;
+
+	CopyUtf8(player._pName, packed.pName, sizeof(player._pName));
+
+	InitPlayer(player, true);
+
+	player._pBaseStr = std::min<uint8_t>(packed.pBaseStr, player.GetMaximumAttributeValue(CharacterAttribute::Strength));
+	player._pStrength = player._pBaseStr;
+	player._pBaseMag = std::min<uint8_t>(packed.pBaseMag, player.GetMaximumAttributeValue(CharacterAttribute::Magic));
+	player._pMagic = player._pBaseMag;
+	player._pBaseDex = std::min<uint8_t>(packed.pBaseDex, player.GetMaximumAttributeValue(CharacterAttribute::Dexterity));
+	player._pDexterity = player._pBaseDex;
+	player._pBaseVit = std::min<uint8_t>(packed.pBaseVit, player.GetMaximumAttributeValue(CharacterAttribute::Vitality));
+	player._pVitality = player._pBaseVit;
+	player._pStatPts = packed.pStatPts;
+
+	player._pExperience = SDL_SwapLE32(packed.pExperience);
+	player._pGold = SDL_SwapLE32(packed.pGold);
+	player._pBaseToBlk = PlayersData[static_cast<std::size_t>(player._pClass)].blockBonus;
+	if ((int)(player._pHPBase & 0xFFFFFFC0) < 64)
+		player._pHPBase = 64;
+
+	player._pMaxManaBase = SDL_SwapLE32(packed.pMaxManaBase);
+	player._pManaBase = SDL_SwapLE32(packed.pManaBase);
+	player._pManaBase = std::min<int32_t>(player._pManaBase, player._pMaxManaBase);
+	player._pMemSpells = SDL_SwapLE64(packed.pMemSpells);
+
+	for (int i = 0; i < 37; i++) // Should be MAX_SPELLS but set to 36 to make save games compatible
+		player._pSplLvl[i] = packed.pSplLvl[i];
+	for (int i = 37; i < 47; i++)
+		player._pSplLvl[i] = packed.pSplLvl2[i - 37];
+
+	bool isHellfire = packed.bIsHellfire != 0;
+
+	for (int i = 0; i < NUM_INVLOC; i++)
+		UnPackItem(packed.InvBody[i], player, player.InvBody[i], isHellfire);
+
+	player._pNumInv = packed._pNumInv;
+	for (int i = 0; i < player._pNumInv; i++)
+		UnPackItem(packed.InvList[i], player, player.InvList[i], isHellfire);
+
+	for (int i = 0; i < InventoryGridCells; i++)
+		player.InvGrid[i] = packed.InvGrid[i];
+
+	VerifyGoldSeeds(player);
+
+	for (int i = 0; i < MaxBeltItems; i++)
+		UnPackItem(packed.SpdList[i], player, player.SpdList[i], isHellfire);
+
+	CalcPlrInv(player, false);
+	player.wReflections = SDL_SwapLE16(packed.wReflections);
+	player.pDiabloKillLevel = SDL_SwapLE32(packed.pDiabloKillLevel);

```

Comment:
```suggestion
	player.pDiabloKillLevel = SDL_SwapLE32(packed.pDiabloKillLevel);
	player.pDamAcFlags = static_cast<ItemSpecialEffectHf>(SDL_SwapLE32(packed.pDamAcFlags));
```

Reply:
Oh I intentionally removed this as it should be calculated by `CalcPlrInv()` and so not needed. Is that not correct?

Reply:
Oh, I get it. I just thought it was an oversight.

Reply:
I should probably have made it more clear.